data_juicer/ops/deduplicator/ray_bts_minhash_deduplicator.py

```diff
@@ -549,22 +641,40 @@ def minhash_with_uid(table: pa.Table) -> pa.Table:
 
         tmp_dir = os.path.join(self.work_dir, '.tmp',
                                ray.get_runtime_context().get_job_id())
-        dataset.map_batches(
-            minhash_with_uid,
-            batch_format='pyarrow',
-            zero_copy_batch=True,
-        ).write_parquet(tmp_dir)
-        dataset = ray.data.read_parquet(tmp_dir)
+        if self.use_gpu:
+            dataset = dataset.map_batches(
+                GPUMinHashActor,
+                batch_format='pyarrow',
+                zero_copy_batch=True,
+                num_gpus=1,
+                concurrency=3,
+                batch_size=self.minhash_batch_size,
+            )
+            dataset.map_batches(
```

Comment:
Is there any way to merge these two `map_batches` into one? Adding additional `map_batches` may increase network overhead.